integrations/octopus/src/parser.ts

```diff
@@ -0,0 +1,21 @@
+export const parseBlockByNumber = async (
+  blockNumber: number,
+  api: any,
+  client: any
+): Promise<{ block: any; tags: any }> => {
+  const hash = await api.rpc.chain.getBlockHash(blockNumber);
+
+  const result = await client.request({
+    method: "chain_getBlock",
+    params: [hash.toHex()],
+  });
+  let block = JSON.stringify(result);
+
+  const tags = [
+    { name: "ChaidId", value: 9999 },
```

Comment:
@baidang201 Do we want to hardcode `9999`?

Reply:
we can init from process.env.CHAINID, and init instances.push(OctopusInstance(poolID, ChainId, stake, wallet));

what do you think

Reply:
@baidnag201 Is it possible to pull that through the endpoint?

Reply:
api.rpc.GetStorage(StorageKey('octopusAppachain', 'AppchainId')) 
but it not a Standard function in substrate

i think we can put it in .env, that we can Compatible with substrate

Reply:
@baidang201 I'd prefer if we didn't put it in the `.env` because some people might forget to do that etc.

After a quick google, found the following [here](https://polkadot.js.org/docs/substrate/rpc/#eth):

<img width="548" alt="Screenshot 2021-07-17 at 09 39 09" src="https://user-images.githubusercontent.com/62398724/126031472-3466176e-011a-4961-bf8d-e55e4bb33882.png">

Reply:
i had try api.rpc.eth.chainId() in https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fpolkadot.elara.patract.io#/js

it report s "api.rpc.eth.chainId is not a function"

it is not a standard function call in every substrate base chain.

Reply:
how about  change .env file to  k8s or docker compose env, that they would not forget.

Reply:
@baidang201 Does the chain ID change per Octopus endpoint?

If yes, we shouldn't use a `.env` file, because if the node operator is running multiple pools at once, they can't have multiple environment variables per pool 😅

Reply:
yes, every  Octopus endpoint chain ID will be different.
i agree use rpc call first, 
const t = await api.query.octopusAppchain.appchainId();

i will add api.query.octopusAppchain.appchainId();  in code later.

Reply:
every Octopus endpoint's  chanid is different,

i think use  api.query.octopusAppchain.appchainId() get chainId  is a good choose.

Reply:
@baidang201 So I ran the following:

```js
(await api.query.octopusAppchain.appchainId()).toHuman()
```

And I get `easydeal-demo`, which is the exact same return as running:

```js
await api.rpc.system.name()
```

I would've assume the ChainID is a number? 🤔

Reply:
it is a string by design